BetaCameras/RealSense2/RealSense2.cs

```diff
@@ -1458,76 +1417,59 @@ protected override void UpdateImpl()
 
             while (true)
             {
-                RealSense2API.RS2Frame data = RealSense2API.PipelineWaitForFrames(_pipeline, 5000);
-
-                if (!data.IsValid() || data.Handle == IntPtr.Zero)
-                {
-                    RealSense2API.ReleaseFrame(data);
-                    continue;
-                }
+                FrameSet data = _pipeline.WaitForFrames(5000);
 
-                int frameCount = RealSense2API.FrameEmbeddedCount(data);
+                int frameCount = data.Count;
                 log.Debug(string.Format("RealSense2: Got {0} Frames", frameCount));
```

Comment:
Please remove (or comment) this and all per-frame (i.e. in Update and CalcChannel) log outputs.
They tend to fill log files very quickly.

Reply:
This is unrelated to the original change and can therefore become an issue.

Reply:
fixed